src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -634,5 +1012,72 @@ internal bool TryGetSpanOnHigherBuffer(SnapshotSpan snapshotSpan, ITextBuffer ta
             span = default;
             return false;
         }
+
+        [NonCopyable]
+        private struct State
+        {
+            /// <summary>
+            /// The current expansion session.
+            /// </summary>
+            public IVsExpansionSession _expansionSession;
+
+            /// <summary>
+            /// The set of symbols initially identified as candidates for providing arguments. When a snippet is
+            /// constructed with parameters for a specific symbol, <see cref="_symbol"/> will identify the specific
+            /// symbol from within this collection matching the current session.
+            /// </summary>
+            public ImmutableArray<IMethodSymbol> _symbols;
+
+            /// <summary>
+            /// The current symbol presented in an Argument Provider snippet session. This may be null if Signature Help
+            /// has not yet provided a symbol to show.
+            /// </summary>
+            public IMethodSymbol _symbol;
```

Comment:
_method or _currentMethod or _currentSelectedMethod.  I personally think the latter two read better.

Reply:
so, 

> This may be null if Signature Help has not yet provided a symbol to show.

This makes a lot of sense.  We're relying on sighelp to tell us which method a user is looking at.  However, this strongly indicates to me that we should also rely on sighelp to tell us the set of methods in the sighelp group.

Reply:
⏱ updated to clarify that `_methods` is the set of symbols the session was _originally_ created with (prior to Signature Help starting).